compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7024,6 +7029,28 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void populateIncludedRecordParams(BVarSymbol param, List<BRecordType> incRecords) {
+        if (Symbols.isFlagOn(param.flags, Flags.INCLUDED)) {
+            incRecords.add((BRecordType) Types.getReferredType(param.type));
+        }
+    }
+
+    // If there is a named-arg or positional-arg corresponding to an included-record-param,
+    // it is an error for a named-arg to specify a field of that included-record-param.
+    private void checkSameNamedArgsInIncRecords(List<BLangExpression> namedArgs, List<BRecordType> incRecords) {
+        for (BLangExpression namedArg : namedArgs) {
+            String argName = ((NamedArgNode) namedArg).getName().value;
+            for (BRecordType record : incRecords) {
+                if (record.fields.containsKey(argName)) {
+                    dlog.error(namedArg.pos,
+                            DiagnosticErrorCode.NAMED_ARG_NOT_ALLOWED_INCLUDED_IN_INCLUDED_RECORD_PARAM, argName,
+                            record.tsymbol.name);
```

Comment:
Consider the below example
```ballerina
type Configuration record {|
    int id?;
    string name = "";
|};

function fn(*Configuration config) {

}

public function main() {
    fn(id = 1, config = {name: "default"});
    // fn(id = 1, name = "");
    // fn({name: "default"}, id = 1);
}

```
With the current approach, we give the error to `id=1`. IMO actual error with the ` config = {name: "default"}`.
Any thoughts @MaryamZi

Reply:
@chiranSachintha, is your concern the same as https://github.com/ballerina-platform/ballerina-lang/pull/35800#discussion_r874452262? That we need to try and fix to give for the second occurrence onward.

I think your particular example is an interesting case, because the argument for `config` doesn't actually specify a value for `id`. But we don't always know this since the argument for `config` may be a variable reference.

```ballerina
    Configuration config = {name: "default"};
    fn(id = 1, config = config);
```

I think it's OK for it to be an error, but may be we should reword the error for optional fields. E.g.,
```cmd
duplicate arguments may have been specified for 'id' via the named argument and the argument for the included record parameter
```

Reply:
If the argument for `config` doesn't actually specify a value for `id` then the value for `id` will be nil right? so it's anyway going to duplicate `id=1`?

Reply:
No, `id` is an optional field (rather than a nilable field with nil as the default value, in which case the value will be nil).

Reply:
Changed the error message

Reply:
Seems like you have changed it for the non-optional case also.

```ballerina
type Foo record {
    int id;
    string name;
};

function fn(*Foo record1) {

}

public function main() {
    fn(id = 2, record1 = {id: 2, name: ""});
}
```

Here, `id` is definitely duplicated. So saying "duplicate arguments may have been specified" isn't quite right.

Reply:
Both optional case and non-optional case both fall into the following sentence in the spec right?
```
If there is a named-arg or positional-arg corresponding to an included-record-param, it is an error for a named-arg to specify a field of that included-record-param. 
```
Therefore error messages should be same right?

Reply:
Not necessarily. We need to make sure we are properly communicating what the issue is to the user.

Reply:
Okay, For this case, you mean having the word `may` is not necessary in `duplicate arguments may have been specified` right?

Reply:
Yes. Basically have two different errors for the optional and non-optional case.

Or we can actually say something like

```
cannot specify a named argument for a field of an included record parameter when an argument is specified for the included record parameter
```

This is more in-line with the spec. The errors have to be for the arguments for the fields.

Reply:
Changed